code/obj/machinery/dialysis.dm

```diff
@@ -0,0 +1,169 @@
+TYPEINFO(/obj/machinery/dialysis)
+	mats = list("MET-1" = 20, "CRY-1" = 5, "CON-2" = 5)
+
+/obj/machinery/dialysis
+	name = "dialysis machine"
+	desc = "A machine which continuously draws blood from a patient, removes excess chemicals from it, and re-infuses it into the patient."
+	icon = 'icons/obj/machines/dialysis.dmi'
+#ifdef IN_MAP_EDITOR
+	icon_state = "dialysis-map"
+#else
+	icon_state = "dialysis-base"
+#endif
+	density = 1
+	var/mob/living/carbon/patient
+	var/list/whitelist
+	// In units per process tick.
+	var/draw_amount = 10
+	var/hacked = FALSE
+	var/last_in = 0
+	var/last_out = 0
+	var/output_blood_colour
+	/// Reagent ID of the current patient's blood.
+	var/patient_blood_id
+
+	New()
+		..()
+		src.create_reagents(100)
+		if (islist(chem_whitelist) && length(chem_whitelist))
```

Comment:
pointless check as it defaults to null anyway, if you want to check for this failure state then CRASH or stack trace instead